py/foundation/geometry.py

```diff
@@ -95,17 +100,21 @@ def intersection(Is: Iterable['Interval']) -> Optional['Interval']:
         'of an empty list of intervals')
     return Interval.build(max(I.a for I in Is), min(I.b for I in Is))
 
-  def dump(self) -> Dict:
+  def as_dict(self) -> Dict:
     return {
       'a': self.a,
       'b': self.b,
     }
 
-
-@dataclass(frozen=True)
-class Point:
+@dataclass
+class Point(FoundationType):
   x: float
   y: float
+  page_index: int
+
+  @property
+  def type(self) -> str:
+    return "Point"
 
   def __str__(self) -> str:
     return "Point({}, {})".format(self.x, self.y)
```

Comment:
This is missing `page_index`.

Reply:
Will implement the previous comment instead